### DISCUSSION TITLE
Is there any target date for the new release?

### DISCUSSION THREAD

**Discussion by JoseAmador95** [ThrowTheSwitch/Ceedling#825] [General]
The latest stable release of Ceedling is over 2 years old and is no longer compatible (out-of-the-box) with newer versions of Ruby. I like Ceedling, and these issues are not critical today as I use Ubuntu 22.04 LTS. However, I expect Ceedling to be broken in the newer Ubuntu versions due to the version of Ruby that will come in the distro.

**Comment by mvandervoord** [comment]
We are working towards a release at the end of this month.

**Comment by rdlepage** [comment]
Please give an update on progress toward the next stable release (I assume 0.32.0). 
Thanks.

**Comment by JoseAmador95** [comment]
Any news about the release?

**Comment by mkarlesky** [comment]
The journey has been winding and involved. We are sorry for how long this has all taken! 0.32 is actually no more. The next release will be 1.0.0!

All the major planned work and documentation is now done. Some final testing, small issue resolution, and packaging work is now happening. If all goes well, 1.0.0 could be published this month (July of 2024).

**Reply by jmrubillon** [reply]
Any update on when we can expect 1.0.0? I know you were hoping July 2024, we're in October now so life has happened and we're just wondering as ceelding is broken by the newer versions of Ruby :(

**Reply by mkarlesky** [reply]
@jmrubillon Documentation is done. All planned feature work is complete. The various installation packages, including Docker images, are ready. We've been hunting down a difficult to reproduce threading / process spawning problem that is mostly showing up in Windows (when it shows up). We have it mostly resolved but are still trying to sort out some missing error reporting. This is quite literally the last thing. As you can imagine, bug hunting like this is challenging work to estimate. We're _this close_. Sigh. So sorry.

**Reply by jmrubillon** [reply]
@mkarlesky Thanks for the update. I'd offer my services hunting the bug down, but I know nothing about ruby so that would be futile...
Knowing we're this close to having a 1.0.0 is encouraging news.

**Reply by mkarlesky** [reply]
@jmrubillon It looks as though @mvandervoord has found and resolved the problem. It required enough changes and to notoriously complex threading mechanisms to warrant some special testing. We're lining up those who have reported issues that all seem to stem from the same Ceedling bug (pun intended) to exercise the fix. If and when that proves successful, we'll finally promote 1.0.0 to production status.

**Comment by iga31** [comment]
How about mentioning in the ReadMe of the main branch that version 1.0.0 is currently under development?
Many people may not know that the latest version is under development.

**Reply by mkarlesky** [reply]
@iga31 This is a great idea. I wish we had thought of it months ago. I will make this change, but it will probably only be useful for an unfortunately brief amount of time as we just cleared the last major hurdle before release (fixing an elusive threading bug that caused a hard to reproduce problem mostly in Windows).

**Comment by JoseAmador95** [comment]
Hi! Is there a target release date? Do you have an issue number for the bug that is blocking the release?

**Comment by mkarlesky** [comment]
Hi, @JoseAmador95. The very last thing is issue #706. It's been a problem for quite a while (the various related issues spider back many years). We think it's an essential ability for complicated projects, and it must be present if the project is (finally) publishing the 1.0.0 version. Unfortunately, as is the case with community-driven projects, delays are common with both the developer contributors and user contributors. Here, at the moment, we're waiting to hear back from a user contributor who has been deeply invested in this issue and has a great, sophisticated project for exercising the issue. _Technically_, I _**think**_ it's fixed. But, we've run into a secondary problem (that I think _may_ have a quick solution). Waiting to hear back on all this. Otherwise, apart from some very minimal final packaging work, everything—docs, code, tests, etc.—is ready to be pushed out.

**Reply by jmrubillon** [reply]
Could we get 1.0.0 as a Christmas present? Nicely wrapped under the tree?
Please, Santa, we've all been good.

**Reply by LuisAfonso95** [reply]
#706 working would be huge if I am reading it correctly! 
Looking forward to migrating to a full release (we're using a random snapshot that worked and haven't bothered updating)

**Reply by mkarlesky** [reply]
@jmrubillon It wasn't Christmas, but Happy New Year. 1.0.0 was released on January 1!

**Reply by jmrubillon** [reply]
@mkarlesky It's a lovely start to the new year. Thank you to all the contributors for their efforts to bring us 1.0.0.
